compysition/actors/eventgenerator.py

```diff
@@ -195,9 +195,9 @@ def __init__(self, name, *args, **kwargs):
         ScheduledUDPEventProducer._init_peers_interface(self, *args, **kwargs)
 
     def _do_produce(self):
-        if not self._is_running:
+        if not self._is_running and self.peers_interface.is_master():
             self._is_running = True
-            super(ScheduledUDPEventProducer, self)._do_produce()
+            EventProducer._do_produce(self)
```

Comment:
What's getting rid of `super` doing here? Is that necessary as part of the fix? I think it's generally better to let `super` find the correct method in the class hierarchy, just wondering why that might not be the case here.

Reply:
Two reasons ...

- super() would end up calling UDPEventProducer._do_produce which (though technically functional) would add unnecessary/redundant execution vs directly calling EventProducer._do_produce.
- super changes slightly from python2 to python3.  And it is my understanding that the best practice for making code compatible with both is to use this format for inheritance references.